stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -120,25 +122,28 @@ Iterating the decomposition produces the components `L` and `U`.
 
 # Examples
 ```jldoctest
-julia> A = [4. 12. -16.; 12. 37. -43.; -16. -43. 98.]
-3×3 Matrix{Float64}:
-   4.0   12.0  -16.0
-  12.0   37.0  -43.0
- -16.0  -43.0   98.0
+julia> X = [1.0, 2.0, 3.0, 4.0];
 
-julia> C = cholesky(A, Val(true))
+julia> A = X * X';
+
+julia> C = cholesky(A, Val(true), check = false)
 CholeskyPivoted{Float64, Matrix{Float64}}
-U factor with rank 3:
-3×3 UpperTriangular{Float64, Matrix{Float64}}:
- 9.89949  -4.34366  -1.61624
-  ⋅        4.25825   1.1694
-  ⋅         ⋅        0.142334
+U factor with rank 1:
+4×4 UpperTriangular{Float64, Matrix{Float64}}:
+ 4.0  2.0  3.0  1.0
+  ⋅   0.0  6.0  2.0
+  ⋅    ⋅   9.0  3.0
+  ⋅    ⋅    ⋅   1.0
```

Comment:
Values outside the first line are completely arbitrary, right? I wonder whether it would be clearer to set them to zero.

Reply:
Outside the first line of what? You commented on `U`, but you meant to manipulate `A` or `X`? Did you mean something like
```julia
julia> A = [1.0 2.0 3.0; 2.0 0.0 0.0; 3.0 0.0 0.0]
3×3 Matrix{Float64}:
 1.0  2.0  3.0
 2.0  0.0  0.0
 3.0  0.0  0.0

julia> cholesky(A, Val(true), check = false)
CholeskyPivoted{Float64, Matrix{Float64}}
U factor with rank 1:
3×3 UpperTriangular{Float64, Matrix{Float64}}:
 1.0   2.0  3.0
  ⋅   -4.0  0.0
  ⋅     ⋅   0.0
permutation:
3-element Vector{Int64}:
 1
 2
 3
```
I'm confused though about the rank computation here, and sure enough, it gives
```julia
julia> C.U[1:C.rank, :]' * C.U[1:C.rank, :] ≈ A[C.p, C.p]
false
```

Reply:
Sorry I meant the first row of the matrix U. In the example I give in the docstring, only the first row is used, so AFAICT other values in U are arbitrary.

The example you give is worrying. So in that case we have no idea how to get A from C? :-/

Reply:
Ah, actually the problem seems to be that the equality holds only if A is positive semidefinite, which isn't the case of your example. We should probably mention this, just like R does [in a warning section](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/chol).

Reply:
Unfortunately I don't know how to detect whether the matrix is merely rank-deficient, or not definite. [LAPACK docs](http://www.netlib.org/lapack/explore-html/da/dba/group__double_o_t_h_e_rcomputational_ga31cdc13a7f4ad687f4aefebff870e1cc.html) don't say how to distinguish these two reasons of failure. I guess it's just not possible? We just throw a `RankDeficientException` in both cases, and R prints a warning which mixes the two possible reasons.

Reply:
Apparently, we don't even have a show method for that exception. We could add one saying something like "Cholesky decomposition had $(ex.info) successful steps, matrix is either rank deficient or is not positive semidefinite"?

Reply:
> Values outside the first line are completely arbitrary, right? I wonder whether it would be clearer to set them to zero.

I think I only know seem to understand what you were proposing, maybe. Did you mean to first compute the decomposition, and then do something like `F.factors[F.rank+1:end,F.rank+1:end] .= 0`? It would add a little to the workload, though. Not sure how often this would be triggered.

Reply:
Shall we have that discussion in a different issue/PR, and merge this as is? In terms of "Fix pivoted cholesky docstrings", I think this PR succeeded.

Reply:
Yes that's what I mean. Anyway that was mostly out of curiosity. I think we'd better merge this PR first.